src/act.informative.cpp

```diff
@@ -3843,6 +3843,10 @@ ACMD(do_scan) {
     int from_room, to_room;
     Exit *exit;
 
+    if (GET_CLASS(ch) != CLASS_ROGUE) {
+        GET_HIDDENNESS(ch) = 0;
```

Comment:
Why not just set this for all classes?  Are rogues the only ones that can't scan while hidden?